src/saving_tools.jl

```diff
@@ -65,6 +65,20 @@ function gitdescribe(gitpath = projectdir())
     return c
 end
 
+"""
+    gitpatch(gitpath = projectdir())
+
+Generates a patch describing the changes of a dirty repository
+compared to its last commit.
+"""
+function gitpatch(gitpath = projectdir())
+    # tree = LibGit2.GitTree(repo, "HEAD^{tree}")
+    # diff = LibGit2.diff_tree(repo, tree)
+    # now there is no way to generate the patch with LibGit2.jl.
+    # Instead use commands:
+    patch = read(`git --git-dir=$(gitpath)/.git diff `, String)
```

Comment:
Maybe a `git diff HEAD` would be better here to also include staged files

Reply:
I mean, it depends on what you want actually. If you want to diff the index, than you version is fine. However, the docstrings says diff the last commit, so you need `HEAD`.

Reply:
You're right, thanks!